app/models/measure_test.rb

```diff
@@ -85,4 +87,8 @@ def cat3_status
       'unstarted'
     end
   end
+
+  def provider
+    Provider.find(provider_id)
```

Comment:
Can't we' make this an active model relationship, otherwise it will make a db request every time you call .provider